File: sdks/csharp/PgChangeFeed.Client/Nats/PgChangeFeedNatsMalformedMessageException.cs
namespace PgChangeFeed.Client.Nats;

/// <summary>
/// Thrown when a message on the NATS stream does not decode to the ten-field
/// JSON change — the NATS counterpart of
/// <c>PgChangeFeed.Client.Http.PgChangeFeedMalformedResponseException</c>.
///
/// It is its own small exception type, not part of
/// <c>PgChangeFeed.Client.Http.PgChangeFeedException</c>'s hierarchy: that
/// hierarchy's <c>StatusCode</c> is an HTTP status, and a NATS message carries
/// no status code, only a subject and a payload.
/// </summary>
public sealed class PgChangeFeedNatsMalformedMessageException : Exception
{
    /// <summary>Creates the exception with a description of the problem.</summary>
    public PgChangeFeedNatsMalformedMessageException(string message)
        : base(message)
    {
    }

    /// <summary>Creates the exception with a description of the problem and the cause.</summary>
    public PgChangeFeedNatsMalformedMessageException(string message, Exception innerException)
        : base(message, innerException)
    {
    }
}

File: sdks/kotlin/pgchangefeed-kotlin/src/main/kotlin/io/github/pt9912/pgchangefeed/nats/PgChangeFeedNatsMalformedMessageException.kt
package io.github.pt9912.pgchangefeed.nats

/**
 * Thrown when a message on the NATS stream does not decode to the ten-field
 * JSON change — the NATS counterpart of
 * [io.github.pt9912.pgchangefeed.http.PgChangeFeedMalformedResponseException].
 *
 * It is its own small exception type, not part of
 * [io.github.pt9912.pgchangefeed.http.PgChangeFeedException]'s hierarchy: that
 * hierarchy's `statusCode` is an HTTP status, and a NATS message carries no
 * status code, only a subject and a payload.
 */
class PgChangeFeedNatsMalformedMessageException(
    message: String,
    cause: Throwable? = null,
) : Exception(message, cause)
